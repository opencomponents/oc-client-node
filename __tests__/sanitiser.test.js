'use strict';

const sanitiser = require('../src/sanitiser');

test('sanitiser api', () => {
  expect(sanitiser).toMatchSnapshot();
});

test('sanitiseConfiguration', () => {
  const result = sanitiser.sanitiseConfiguration();
  expect(result).toMatchSnapshot();
});

test('sanitiseGlobalGetInfoOptions', () => {
  const result = sanitiser.sanitiseGlobalGetInfoOptions();
  expect(result).toMatchSnapshot();
});

describe('sanitiseGlobalRenderOptions', () => {
  const scenarios = [
    {
      description: 'all empty',
      options: {},
      config: {}
    },
    {
      description: 'clientRendering is false',
      options: {},
      config: { registries: { clientRendering: false } }
    },
    {
      description: 'options is a function',
      options: jest.fn(),
      config: {}
    },
    {
      description: 'options is a function',
      options: jest.fn(),
      config: {}
    },
    {
      description: 'options w/ headers',
      options: { headers: { 'X-OT-Channel': 'MobileWeb' } },
      config: {}
    },
    {
      description: 'container is true and renderInfo is false',
      options: { container: true, renderInfo: false },
      config: {}
    }
  ];

  scenarios.forEach(({ description, options, config }) => {
    test(description, () => {
      const result = sanitiser.sanitiseGlobalRenderOptions(options, config);
      expect(result).toMatchSnapshot();
    });
  });
});
