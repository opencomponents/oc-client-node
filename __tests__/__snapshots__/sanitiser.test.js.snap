// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sanitiseConfiguration 1`] = `
Object {
  "cache": Object {},
  "components": Object {},
}
`;

exports[`sanitiseGlobalGetInfoOptions 1`] = `
Object {
  "headers": Object {
    "accept": "application/vnd.oc.info+json",
    "user-agent": "oc-client-2.1.21/v8.2.1-darwin-x64",
  },
  "timeout": 5,
}
`;

exports[`sanitiseGlobalRenderOptions all empty 1`] = `
Object {
  "container": false,
  "headers": Object {
    "accept": "application/vnd.oc.unrendered+json",
    "user-agent": "oc-client-2.1.21/v8.2.1-darwin-x64",
  },
  "renderInfo": true,
  "timeout": 5,
}
`;

exports[`sanitiseGlobalRenderOptions clientRendering is false 1`] = `
Object {
  "container": false,
  "disableFailoverRendering": true,
  "headers": Object {
    "accept": "application/vnd.oc.unrendered+json",
    "user-agent": "oc-client-2.1.21/v8.2.1-darwin-x64",
  },
  "renderInfo": true,
  "timeout": 5,
}
`;

exports[`sanitiseGlobalRenderOptions container is true and renderInfo is false 1`] = `
Object {
  "container": true,
  "headers": Object {
    "accept": "application/vnd.oc.unrendered+json",
    "user-agent": "oc-client-2.1.21/v8.2.1-darwin-x64",
  },
  "renderInfo": false,
  "timeout": 5,
}
`;

exports[`sanitiseGlobalRenderOptions options is a function 1`] = `
Object {
  "container": false,
  "headers": Object {
    "accept": "application/vnd.oc.unrendered+json",
    "user-agent": "oc-client-2.1.21/v8.2.1-darwin-x64",
  },
  "renderInfo": true,
  "timeout": 5,
}
`;

exports[`sanitiseGlobalRenderOptions options is a function 2`] = `
Object {
  "container": false,
  "headers": Object {
    "accept": "application/vnd.oc.unrendered+json",
    "user-agent": "oc-client-2.1.21/v8.2.1-darwin-x64",
  },
  "renderInfo": true,
  "timeout": 5,
}
`;

exports[`sanitiseGlobalRenderOptions options w/ headers 1`] = `
Object {
  "container": false,
  "headers": Object {
    "accept": "application/vnd.oc.unrendered+json",
    "user-agent": "oc-client-2.1.21/v8.2.1-darwin-x64",
    "x-ot-channel": "MobileWeb",
  },
  "renderInfo": true,
  "timeout": 5,
}
`;

exports[`sanitiser api 1`] = `
Object {
  "sanitiseConfiguration": [Function],
  "sanitiseGlobalGetInfoOptions": [Function],
  "sanitiseGlobalRenderOptions": [Function],
}
`;
